NinchatSDKSwift/NINChatSession.swift

```diff
@@ -45,183 +40,96 @@ public final class NINChatSessionSwift: NINChatSession, NINChatSessionProtocol {
     
     // MARK: - NINChatSessionClosure
     
-    public var didOutputSDKLog: ((NINChatSession, String) -> Void)?
-    public var onLowLevelEvent: ((NINChatSession, NINLowLevelClientProps, NINLowLevelClientPayload, Bool) -> Void)?
-    public var overrideImageAsset: ((NINChatSession, AssetConstants) -> UIImage?)?
-    public var overrideColorAsset: ((NINChatSession, ColorConstants) -> UIColor?)?
-    public var didEndSession: ((NINChatSession) -> Void)?
+    public var didOutputSDKLog: ((NINChatSessionSwift, String) -> Void)?
+    public var onLowLevelEvent: ((NINChatSessionSwift, NINLowLevelClientProps, NINLowLevelClientPayload, Bool) -> Void)?
+    public var overrideImageAsset: ((NINChatSessionSwift, AssetConstants) -> UIImage?)?
+    public var overrideColorAsset: ((NINChatSessionSwift, ColorConstants) -> UIColor?)?
+    public var didEndSession: ((NINChatSessionSwift) -> Void)?
     
     // MARK: - NINChatSessionProtocol
     
-    public weak var delegateSwift: NINChatSessionDelegateSwift?
-    override public var serverAddress: String? {
-        set {
-            self.sessionManager.serverAddress = newValue
-        }
-        get {
-            return self.sessionManager.serverAddress ?? defaultServerAddress
-        }
-    }
-    override public var siteSecret: String? {
-        set {
-            self.sessionManager.siteSecret = newValue
+    public weak var delegate: NINChatSessionDelegateSwift?
+    public var appDetails: String? {
+        didSet {
+            sessionManager.appDetails = appDetails
         }
-        get {
-            return self.sessionManager.siteSecret
-        }
-    }
-    override public var audienceMetadata: NINLowLevelClientProps? {
-        set {
-            self.sessionManager.audienceMetadata = newValue
-        }
-        get {
-            return self.sessionManager.audienceMetadata
-        }
-    }
-    override public var session: NINLowLevelClientSession {
-        guard self.started else {
-            fatalError("API has not been started")
-        }
-        return self.sessionManager.session
     }
     
-
-    public init(configKey: String, queueID: String?, environments: [String]?) {
+    public init(configKey: String, queueID: String? = nil, environments: [String]? = nil, serverAddress: String? = nil, siteSecret: String? = nil, metadata: NINLowLevelClientProps? = nil) {
```

Comment:
Since app developers never change the server address or set a site secret, they should not clutter the primary method signatures. Would it be possible to make an alternative init method which is for our use only?

I would like to avoid any confusion about the necessity of configuring server address and site secret, by moving them out of sight.

It would be best if they could be configured via standalone setter methods instead of an initialization method, unless that's impossible (= if the session which needs them is created by the init method).

Reply:
If they have to be hidden from developers, could we not expose them at all? They can be some private variables within the SDK.

Reply:
I removed parameters from the initializer and made corresponded variables for any internal usages.

Reply:
They don't need to be *hidden*, just out of the way to avoid confusion and unnecessary questions.

Reply:
Yes, I changed them to variables that can be specified/skipped independently.